python/cugraph/community/leiden.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2019 - 2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.community import leiden_wrapper
+from cugraph.structure.graph import Graph
+
+
+def leiden(input_graph, max_iter=100, resolution=1.):
+    """
+    Compute the modularity optimizing partition of the input graph using the
+    Louvain heuristic
+
+    Parameters
+    ----------
+    input_graph : cugraph.Graph
+        cuGraph graph descriptor of type Graph
+
+        The adjacency list will be computed if not already present. The graph
```

Comment:
"where an undirected edge is represented by a directed edge in both direction." : this is now taken care of in the `Graph` class. User doesn't need to worry or know about this